packages/data-context/src/data/ProjectConfigManager.ts

```diff
@@ -153,8 +154,84 @@ export class ProjectConfigManager {
     if (this._registeredEventsTarget && this._testingType !== this._registeredEventsTarget) {
       this.options.refreshLifecycle().catch(this.onLoadError)
     } else if (this._eventsIpc && !this._registeredEventsTarget && this._cachedLoadConfig) {
-      this.setupNodeEvents(this._cachedLoadConfig).catch(this.onLoadError)
+      this.setupNodeEvents(this._cachedLoadConfig)
+      .then(() => {
+        if (this._testingType === 'component') {
+          this.checkDependenciesForComponentTesting()
+        }
+      })
+      .catch(this.onLoadError)
+    }
+  }
+
+  checkDependenciesForComponentTesting () {
+    // if it's a function, for example, the user is created their own dev server,
+    // and not using one of our presets. Assume they know what they are doing and
+    // what dependencies they require.
+    if (typeof this._cachedLoadConfig?.initialConfig?.component?.devServer !== 'object') {
+      return
+    }
+
+    const devServerOptions = this._cachedLoadConfig.initialConfig.component.devServer
+
+    const bundler = WIZARD_BUNDLERS.find((x) => x.type === devServerOptions.bundler)
+
+    // Use a map since sometimes the same dependency can appear in `bundler` and `framework`,
+    // for example webpack appears in both `bundler: 'webpack', framework: 'react-scripts'`
+    const unsupportedDeps = new Map<DependencyToInstall['dependency']['type'], DependencyToInstall>()
+
+    if (!bundler) {
+      return
+    }
+
+    const result = inPkgJson(bundler, this.options.projectRoot)
+
+    if (!result.satisfied) {
+      unsupportedDeps.set(result.dependency.type, result)
+    }
+
+    const isFrameworkSatisfied = (bundler: typeof WIZARD_BUNDLERS[number], framework: typeof WIZARD_FRAMEWORKS[number]) => {
+      for (const dep of framework.dependencies(bundler.type, this.options.projectRoot)) {
+        const res = inPkgJson(dep.dependency, this.options.projectRoot)
+
+        if (!res.satisfied) {
+          return false
+        }
+      }
+
+      return true
+    }
+
+    const frameworks = WIZARD_FRAMEWORKS.filter((x) => x.configFramework === devServerOptions.framework)
+
+    const mismatchedFrameworkDeps = new Map<typeof WIZARD_DEPENDENCIES[number]['type'], DependencyToInstall>()
+
+    let isSatisfied = false
+
+    for (const framework of frameworks) {
+      if (isFrameworkSatisfied(bundler, framework)) {
+        isSatisfied = true
+        break
```

Comment:
Would an early return make sense vs the break?

Reply:
It wouldn't, because even if the framework is ok (eg, CRA), it's possible we want to warn about a mismatches bundler version (as unlikely as this seems, it's possible someone had a supported CRA version, but managed to install a old or new version of Webpack that we don't support).

Reply:
Forgot about the bundler, makes sense!